app/models/submission.rb

```diff
@@ -38,9 +38,12 @@ def submission_template
   end
 
   def new_job
-    Job.new(
+    clstr = attributes['cluster']
+    Job.create(
       submission_id: attributes['id'],
-      cluster: attributes['cluster']
+      cluster: clstr,
+      job_id: SecureRandom.random_number(1000000).to_s + '.' + clstr + '-never.submitted',
```

Comment:
Why does a job that is not submitted need a job_id? Why can't this just be nil until the job is submitted?